music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -0,0 +1,308 @@
+"""SiriusXM Music Provider for Music Assistant."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator, Awaitable, Sequence
+from typing import TYPE_CHECKING, Any, cast
+
+from music_assistant.common.models.config_entries import (
+    ConfigEntry,
+    ConfigValueOption,
+    ConfigValueType,
+)
+from music_assistant.common.models.enums import (
+    ConfigEntryType,
+    ContentType,
+    LinkType,
+    MediaType,
+    ProviderFeature,
+    StreamType,
+)
+from music_assistant.common.models.errors import LoginFailed, MediaNotFoundError
+from music_assistant.common.models.media_items import (
+    AudioFormat,
+    ImageType,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemLink,
+    MediaItemType,
+    ProviderMapping,
+    Radio,
+)
+from music_assistant.common.models.streamdetails import StreamDetails
+from music_assistant.server.helpers.webserver import Webserver
+from music_assistant.server.models.music_provider import MusicProvider
+
+if TYPE_CHECKING:
+    from music_assistant.common.models.config_entries import ProviderConfig
+    from music_assistant.common.models.provider import ProviderManifest
+    from music_assistant.server import MusicAssistant
+    from music_assistant.server.models import ProviderInstanceType
+
+import sxm.http
+from sxm import SXMClientAsync
+from sxm.models import QualitySize, RegionChoice, XMChannel
+
+CONF_SXM_USERNAME = "sxm_email_address"
+CONF_SXM_PASSWORD = "sxm_password"
+CONF_SXM_REGION = "sxm_region"
+CONF_SXM_PROXY_PORT = "sxm_proxy_port"
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    return SiriusXMProvider(mass, manifest, config)
+
+
+async def get_config_entries(
+    mass: MusicAssistant,
+    instance_id: str | None = None,
+    action: str | None = None,
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+    return (
+        ConfigEntry(
+            key=CONF_SXM_USERNAME,
+            type=ConfigEntryType.STRING,
+            label="Username",
+            required=True,
+        ),
+        ConfigEntry(
+            key=CONF_SXM_PASSWORD,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Password",
+            required=True,
+        ),
+        ConfigEntry(
+            key=CONF_SXM_REGION,
+            type=ConfigEntryType.STRING,
+            default_value="US",
+            options=tuple(ConfigValueOption(x, x) for x in ("US", "CA")),
```

Comment:
Maybe make this a bit prettier with an actual value and title ?
So:
ConfigValueOption(title="United States", value="US")